src/viam/sdk/common/utils.cpp

```diff
@@ -55,5 +55,21 @@ std::string bytes_to_string(std::vector<unsigned char> const& b) {
     return img_string;
 };
 
+std::chrono::duration<int64_t, std::micro> from_proto(const google::protobuf::Duration& proto) {
+    return std::chrono::microseconds(
+        (int64_t)std::ceil(proto.seconds() * 1e6 + proto.nanos() / 1e3));
+}
+
+google::protobuf::Duration to_proto(const std::chrono::duration<int64_t, std::micro>& duration) {
+    google::protobuf::Duration proto;
+    int64_t total_micros = duration.count();
```

Comment:
Again, I think you can achieve these conversions more seamlessly by using duration_cast, and tests.

Reply:
+1. I appreciate the push. I didn't trust chrono duration enough until I added tests. Now more confident in its arithmetic and rounding behavior.

Reply:
https://github.com/viamrobotics/viam-cpp-sdk/pull/96/commits/5e45493c4a18e7fc10e0f05dc87ae44700344fb1